apisix/plugins/openid-connect.lua

```diff
@@ -319,6 +328,24 @@ local function add_access_token_header(ctx, conf, token)
     end
 end
 
+-- Function to split the scope string into a table
+local function split_scopes_by_space(scope_string)
+    local scopes = {}
+    for scope in string.gmatch(scope_string, "%S+") do
```

Comment:
Why not use [ngx.re.split](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/re.md) ?